### PR TITLE
delete low value tests

### PR DESCRIPTION
**Pull request by Maoni0**
CC @jkotas for issue https://github.com/dotnet/runtime/issues/36850

### CODE REVIEW

**Comment by jkotas** [comment]
Do we have any positive tests for AddMemoryPressure/RemoveMemoryPressure APIs once these are deleted? e.g. if I change these APIs to just validate arguments and do nothing, are there any tests that are going to fail?

File: src/coreclr/tests/src/GC/API/GC/AddUsageTest.cs [hunk 1]
```diff
@@ -1,112 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jkotas) [inline src/coreclr/tests/src/GC/API/GC/AddUsageTest.cs:1]: We also have `RemoveUsageTest` in this directory that has very similar structure. Delete it too?

**Comment by Maoni0** [comment]
I don't think any tests would fail. historically we haven't had much testing coverage in these APIs at all; and the testing we had was mostly done by the interop team. unfortunately we haven't had the resource to add tests for AMP/RMP. I'm only deleting these tests so we don't need to spend time looking at bogus issues.

**Comment by jkotas** [comment]
Ok, I see we have tracking issue for this: https://github.com/dotnet/runtime/issues/36796

**Review by jkotas — APPROVED** [review]
LGTM (modulo the nit about RemoveUsageTest).

**Comment by Maoni0** [comment]
new commit to remove the RemoveUsageTest test + its presence in the target file.

**Pull request merged**